### DISCUSSION TITLE
download private videos

### DISCUSSION THREAD

**Discussion by alagha-go** [DzenanJupic/rustube#62] [Q&A]
I wanted to download private videos from my youtube channel. if i wanted to do this with youtube-dl I would provide cookies or a bearer token and it will output  a json with all the necessary headers and cookies to download the videos. I was planning on using that feature on my application mostly the where i will be using the bearer token provided by google to download the video. will this be implemented any time soon?

**Comment by AniruddhaGawali** [comment]
No it can happen Youtube API has restricted the private, Age Restricted and Unlisted Videos
